### PR TITLE
fix: show correct health checker

### PR DESCRIPTION
**Pull request by juzhiyuan**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

- [x] We have nested logics, need to refactor them;
- [x] Set special flags for upstream_id: Custom/None/Real ID
- [x] If we enable health checker, we should see them when viewing resources (Route/Service);

**Related issues**

None

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases? I**n Progress**
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by LiteSun** [comment]
Found a bug here.
1.  select a service(root) then go to route step2.
2. select None in route step2 then go back to the route step1
3. change service to none in route step1 page then go to route step2 page. Then you will see the bug. 
@juzhiyuan 

https://user-images.githubusercontent.com/31329157/115134191-9f8b7500-a040-11eb-8beb-e305234b2099.mov

**Comment by juzhiyuan** [comment]
Yes, I noticed this case in the morning and fixed it just now. When that case happened, the notification will popup & reset the service_id to the previous value, we'd better now change the original data. Please have a try :)

File: web/src/pages/Route/components/Step1/MetaView.tsx [hunk 1]
```diff
@@ -285,27 +285,44 @@ const MetaView: React.FC<RouteModule.Step1PassProps> = ({ disabled, form, isEdit
...
-                  </Select.Option>
-                );
-              })}
-            </Select>
-          </Form.Item>
-        </Col>
-      </Row>
-    </Form.Item>
+    <React.Fragment>
+      <Form.Item label={formatMessage({ id: 'page.route.service' })} tooltip="绑定服务（Service）对象，以便复用其中的配置。">
```
Comment (LiteSun) [inline web/src/pages/Route/components/Step1/MetaView.tsx:289]: i18n

Comment (juzhiyuan) [reply web/src/pages/Route/components/Step1/MetaView.tsx:289]: Hi, this PR will fix this.

https://github.com/apache/apisix-dashboard/pull/1785/files#diff-48bf46c8113258031a14f9204041bb5ff51de312ef61c7686f41ea3499fe2c14R176

**Review by LiteSun — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

File: web/src/components/Upstream/locales/zh-CN.ts [hunk 2]
```diff
@@ -95,4 +94,10 @@ export default {
 
   'component.upstream.fields.checks.passive.unhealthy.timeouts': '超时时间',
   'component.upstream.fields.checks.passive.unhealthy.timeouts.tooltip': '根据被动健康检查的观察，在代理中认为目标不健康的超时次数。',
+
+  'component.upstream.other.none': '不选择（仅在绑定服务时可用）',
```
Comment (liuxiran) [inline web/src/components/Upstream/locales/zh-CN.ts:98]: we will reuse `UpstreamForm` when creating a service

image

so remove `（仅在绑定服务时可用）` may be  more general here or change another remark ^_^

**Review by liuxiran — APPROVED** [review]

File: web/src/components/Upstream/locales/zh-CN.ts (hunk 2 above)
Comment (juzhiyuan) [reply web/src/components/Upstream/locales/zh-CN.ts:98]: sure, I will handle this case in a new PR :)

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**